module/network/RobotCommunication/src/RobotCommunication.cpp

```diff
@@ -138,11 +138,13 @@ namespace module::network {
                 msg->timestamp = NUClear::clock::now();
 
                 // State
-                int penalty_reason = game_state->data.self.penalty_reason;
-                switch (penalty_reason) {
-                    case 0: msg->state = 0; break;
-                    case 1: msg->state = 1; break;
-                    default: msg->state = 2; break;
+                if (game_state) {
```

Comment:
```suggestion
// Check if game state is null or not to avoid seg fault
                if (game_state) {
```

Reply:
or something similar. Otherwise lgtm

Reply:
```suggestion
                // If there is game state information, then process 
                if (game_state) {
```

Reply:
everything in this reaction is optional and needs to be checked, so I don't think it needs much explanation